File: backend/internal/models/zte_provisioning.go
package models

import "github.com/google/uuid"

// ZTEONUIDMode controls whether the OLT assigns an ONU ID or the request supplies it.
type ZTEONUIDMode string

const (
	ZTEONUIDAuto   ZTEONUIDMode = "auto"
	ZTEONUIDCustom ZTEONUIDMode = "custom"
)

// ZTEGPONRegisterRequest contains the identity and Internet service settings for a ZTE GPON ONU.
type ZTEGPONRegisterRequest struct {
	OLTID           uuid.UUID    `json:"olt_id"`
	Card            int          `json:"card"`
	PON             int          `json:"pon"`
	ONUIDMode       ZTEONUIDMode `json:"onu_id_mode"`
	ONUID           int          `json:"onu_id"`
	SerialNumber    string       `json:"serial_number"`
	ONUType         string       `json:"onu_type"`
	UseVEIP         bool         `json:"use_veip"`
	Name            string       `json:"name"`
	Description     string       `json:"description"`
	ServiceEnabled  bool         `json:"service_enabled"`
	VLANMode        string       `json:"vlan_mode"`
	ServiceType     string       `json:"service_type"`
	VLANID          int          `json:"vlan_id"`
	DownloadProfile string       `json:"download_profile"`
	UploadProfile   string       `json:"upload_profile"`
	// WANMode says whether the OLT writes the ONU's WAN over OMCI or leaves it
	// to be set up on the ONT itself; WANIPMode is how that WAN gets its address
	// and only applies to the former.
	WANMode       string `json:"wan_mode"`
	WANIPMode     string `json:"wan_ip_mode"`
	VLANProfile   string `json:"vlan_profile"`
	PPPoEUsername string `json:"pppoe_username"`
	PPPoEPassword string `json:"pppoe_password"`
}

// VLAN tagging modes on the UNI side of an ONU.
const (
	ZTEVLANModeTag   = "tag"
	ZTEVLANModeUntag = "untag"
)

// Service types the form offers. Bridge carries no OMCI WAN at all.
const (
	ZTEServiceInternet = "internet"
	ZTEServiceBridge   = "bridge"
)

// Where the WAN is configured.
const (
	ZTEWANModeWANIP       = "wan_ip"
	ZTEWANModeSetupViaONT = "setup_via_ont"
)

// How an OMCI-configured WAN obtains its address. The OLT's own help lists
// exactly these three.
const (
	ZTEWANIPModePPPoE  = "pppoe"
	ZTEWANIPModeDHCP   = "dhcp"
	ZTEWANIPModeStatic = "static"
)

File: frontend/src/domain/entities/ZteProvisioning.ts
import type { OltModel } from "./Olt";

export type ZteOnuIdMode = "auto" | "custom";
export type ZteVlanMode = "tag" | "untag";
export type ZteServiceType = "internet" | "bridge";
// Where the WAN is configured, and how an OLT-configured one gets its address.
export type ZteWanMode = "wan_ip" | "setup_via_ont";
export type ZteWanIpMode = "pppoe" | "dhcp" | "static";

export interface ZteGPONRegisterRequest {
  oltId: string;
  card: number;
  pon: number;
  onuIdMode: ZteOnuIdMode;
  onuId: number;
  serialNumber: string;
  onuType: string;
  useVeip: boolean;
  name: string;
  description: string;
  serviceEnabled: boolean;
  vlanMode: ZteVlanMode;
  serviceType: ZteServiceType;
  vlanId: number;
  downloadProfile: string;
  uploadProfile: string;
  wanMode: ZteWanMode;
  wanIpMode: ZteWanIpMode | "";
  vlanProfile: string;
  pppoeUsername: string;
  pppoePassword: string;
  confirm: boolean;
}

export interface ZteProvisionTarget {
  oltId: string;
  oltModel?: OltModel | string;
  card: number;
  pon: number;
  serialNumber: string;
  onuId?: number;
  onuType?: string;
  name?: string;
  description?: string;
}

export interface ZteProvisionResponse {
  jobId: string;
  status: string;
  ontId: string;
  onuId: number;
  commands: string[];
}

export interface ZteProvisionJob {
  jobId: string;
  status: "pending" | "running" | "success" | "failed" | "rolled_back";
  ontId?: string;
  errorMessage?: string;
}

// What the server would send, and the ONU ID it would use.
export interface ZteCommandPreviewResult {
  onuId: number;
  commands: string[];
}
